### PR TITLE
Emoji Keyboard

### LINKED ISSUE #2520: Emoji Keyboard
Screen Shot 2020-12-10 at 12 57 39

### PR DESCRIPTION
**Pull request by onurays**
Fixes #2520 

emoji

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailFragment.kt [hunk 1]
```diff
@@ -478,6 +482,23 @@ class RoomDetailFragment @Inject constructor(
...
+    private fun setupEmojiPopup() {
+        EmojiPopup
+                .Builder
+                .fromRootView(rootConstraintLayout)
+                .setKeyboardAnimationStyle(R.style.emoji_fade_animation_style)
+                .setOnEmojiPopupShownListener { composerLayout.composerEmojiButton.setImageResource(R.drawable.ic_keyboard) }
+                .setOnEmojiPopupDismissListener { composerLayout.composerEmojiButton.setImageResource(R.drawable.ic_insert_emoji) }
+                .build(composerLayout.composerEditText)
+                .also {
+                    emojiPopup = it
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailFragment.kt:494]: Why not just affecting the value like this: `val emojiPopup = EmojiPopup.Builder...`?
It seems that `emojiPopup` does not have to be a member of the class.

**Review by bmarty — COMMENTED** [review]
One first remark, I will do some more test

**Comment by bmarty** [comment]
Also can you add en entry in [this file](https://github.com/vector-im/element-android/blob/bc339191eaf3b6d03eeeba079f025464501d11ab/vector/src/main/assets/open_source_licenses.html#L393) please?

**Review by bmarty — APPROVED** [review]
Thanks for the update!

**Comment by erAck** [comment]
WTF does this add an entire emoji library and buttons on its own?

**Comment by github-tomster** [comment]
@'code creating ones'
unfortunately there seems to be an huge info-gap: 
almost **every** android device has
a font named NotoColorEmoji.ttf 🤦‍♂️ 

you are welcome and i am happy helping you out with this 
not so obvious information!

so what is the intension by **not** using the
system NotoColorEmoji.ttf ❓  🤷‍♂️ 

is this about to take away control from the user ❓

**Comment by github-tomster** [comment]
@ freetime-programmers
does a merge makes a bad code a good code? 🤦‍♂️ 
💉

**Pull request merged**